src/idl_gen_js_ts.cpp

```diff
@@ -1283,7 +1312,10 @@ class JsTsGenerator : public BaseGenerator {
               code += "builder, " + GetArgName(field) + ");\n";
           }
 
-          code += "  return " + struct_def.name + ".end" + struct_def.name + "(builder);\n";
+          code += "  return " + struct_def.name + ".end";
+          if (!parser_.opts.js_ts_short_names) 
```

Comment:
Can you wrap this in a function, such that the code can just be e.g. `code += "  return " + struct_def.name + ".end" + Verbose(struct_def) + "(builder);\n";` ?

Reply:
Oh, yes of course.